src/wallet/wallet2.cpp

```diff
@@ -101,8 +101,9 @@ using namespace cryptonote;
 #define CHACHA8_KEY_TAIL 0x8c
 #define CACHE_KEY_TAIL 0x8d
 
-#define UNSIGNED_TX_PREFIX "Electroneum unsigned tx set"
-#define SIGNED_TX_PREFIX "Electroneum signed tx set"
+// Magic number 004 means these payloads are encrypted.
+#define UNSIGNED_TX_PREFIX "Electroneum unsigned tx set \004"
+#define SIGNED_TX_PREFIX "Electroneum signed tx set \004"
```

Comment:
If you're putting the magic version number back into the constant prefix strings, you'll also need to put back the `- 1` on the calculations of `magiclen` on lines 6276 and 6570.

Reply:
Yes that's right.